cicecore/cicedynB/dynamics/ice_dyn_shared.F90

```diff
@@ -1149,12 +1183,37 @@ subroutine seabed_stress_factor_prob (nx_block, ny_block,          &
          endif
       enddo
 
-      do ij = 1, icellu
-         i = indxui(ij)
-         j = indxuj(ij)
-         ! convert quantities to u-location            
-         Tbu(i,j)  = max(Tbt(i,j),Tbt(i+1,j),Tbt(i,j+1),Tbt(i+1,j+1))
-      enddo                     ! ij          
+      select case (trim(grid_system))
+         case('B')
+            do ij = 1, icellu
+               i = indxui(ij)
+               j = indxuj(ij)
+               ! convert quantities to U-location            
+               Tbu(i,j)  = grid_neighbor_max(Tbt, i, j, 'U')
+            enddo                     ! ij          
+         case('CD')
+            if(present(Tbe)    .and. present(TbN)    .and. &
+               present(icelle) .and. present(icelln) .and. &
+               present(indxei) .and. present(indxej) .and. &
+               present(indxni) .and. present(indxnj)) then
+
+                  do ij = 1, icelle
+                     i = indxei(ij)
+                     j = indxej(ij)
+                     ! convert quantities to E-location
+                     TbE(i,j)  = grid_neighbor_max(Tbt, i, j, 'E')
+                  enddo
+                  do ij = 1, icelln
+                     i = indxni(ij)
+                     j = indxnj(ij)
+                     ! convert quantities to N-location
+                     TbN(i,j)  = grid_neighbor_max(Tbt, i, j, 'N')
+                  enddo
+
+            else
+               call abort_ice(subname // ' insufficient number of arguments for grid_system:' // grid_system)
```

Comment:
This error message doesn't make sense to me.  How about ' grid_system value not implemented:'

Reply:
so that abort is not in the  `case default` branch (which is in fact absent here), it's in the `else` branch of the `if(present (Tbe) .and. ...`. So it's checking the case where a user chose `grid_system == 'CD'` but failed to pass all required arguments for `seabed_stress_factor_prob` necessary for the CD grid.

I can also add a `case default` with an abort if `grid_system` is something else than B or CD, but I think we tend to check these kind of errors early in ice_init, so that's why I did not include such a check.